sdk/create-dapp/templates/react-e2e-counter/src/CreateCounter.tsx

```diff
@@ -49,13 +30,39 @@ export function CreateCounter({
         transaction: tx,
       },
       {
-        onSuccess: (result) => {
-          const objectId = result.effects?.created?.[0]?.reference?.objectId;
-          if (objectId) {
-            onCreated(objectId);
-          }
+        onSuccess: ({ digest }) => {
```

Comment:
If you make this an async function, and await the waitForTransaction, you should be able to uses the `isPending` status from the useSignAndExecuteTransaction hook, rather than tracking your own state

Reply:
Addressed. Had to use `isSuccess` as well as `isPending`. Mentioned in previous comment